webapp/src/app.tsx

```diff
@@ -1532,6 +1542,26 @@ export class ProjectView
         }
     }
 
+    onScaleChanged(oldScale: number, newScale: number) {
+        if (this.isTutorial && oldScale !== newScale) {
+            if (this.initialEditorScale === undefined) {
+                this.initialEditorScale = oldScale;
+            }
+
+            const root = document.getElementById("root");
+            if (root && newScale <= this.initialEditorScale) {
+                // Do not shrink the text beyond its initial size.
+                root.style.setProperty("--tutorialFontSize", `${this.tutorialInitialFontSize}rem`);
+            } else if (root) {
+                // Increase font size to match the editor's % zoom.
+                const maxEditorScale = this.editor.getMaxScale();
+                const zoomAmt = (newScale - this.initialEditorScale) / (maxEditorScale - this.initialEditorScale);
+                const newTutorialFontSize = this.tutorialInitialFontSize + (this.tutorialMaxFontSize - this.tutorialInitialFontSize) * zoomAmt;
```

Comment:
Can you explain this math a bit? I'm having a hard time understanding why we want to multiply zoom amount with the difference of `tutorialMaxFontSize` and `tutorialInitialFontSize`.

Reply:
Yeah, this explanation was extremely helpful! Thank you for that clarification!!